tests/jsmarionette/client/marionette-client/lib/marionette/client.js

```diff
@@ -490,34 +470,37 @@
         args[0] = err;
       }
 
-      return callback.apply(this, args);
+      return cb.apply(this, args);
     },
 
     /**
      * Sends request and formats response.
      *
-     *
      * @private
      * @method _sendCommand
      * @chainable
-     * @param {Object} command marionette command.
-     * @param {String} responseKey the part of the response to pass \
-     *                             unto the callback.
-     * @param {Object} callback wrapped callback.
+     * @param {Object} body The body of the Marionette command
+     * @param {Object} cb wrapped callback
+     * @param {String} key optional key in the response to pass
+     *     unto the callback, will return the full object if undefined
      */
-    _sendCommand: function(command, responseKey, callback) {
-      var self = this;
-
+    _sendCommand: function(body, cb, key) {
       try {
-        return this.send(command, function(data) {
-          var value;
-          try {
-            value = self._transformResultValue(data[responseKey]);
-          } catch (e) {
-            console.log('Error: unable to transform marionette response', data);
-          }
-          return self._handleCallback(callback, data.error, value);
-        });
+        return this.send(body, function(data) {
+          var res, err;
+
+          if ("error" in data)
+            err = data;
```

Comment:
Less nesting is better than more nesting. Once you know that you have an error, call `this._handleCallback` with the error. The result doesn't matter if you have an error.